apps/evaluations/views/dataset_views.py

```diff
@@ -80,10 +80,39 @@ def get_queryset(self):
         return EvaluationDataset.objects.filter(team=self.request.team)
 
     def get_form_kwargs(self):
-        return {**super().get_form_kwargs(), "team": self.request.team}
+        kwargs = super().get_form_kwargs()
+        kwargs["team"] = self.request.team
+        kwargs["filter_params"] = FilterParams.from_request(self.request)
+        kwargs["timezone"] = self.request.session.get("detected_tz", None)
+        return kwargs
+
+    def _get_filter_context_data(self):
+        table_url = reverse("evaluations:dataset_sessions_selection_list", args=[self.request.team.slug])
+        return get_filter_context_data(
+            self.request.team,
+            ExperimentSessionFilter.columns(self.request.team),
+            "last_message",
+            table_url,
+            "sessions-table",
+        )
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update(self._get_filter_context_data())
+        return context
+
+    def form_valid(self, form):
+        # Save basic dataset info first
+        response = super().form_valid(form)
+
+        mode = form.cleaned_data.get("mode")
+        if mode == "clone":
+            form.clone_messages_to_dataset(self.object)
```

Comment:
Super nit: Should `clone_messages_to_dataset` not maybe be a standalone function instead that receives whatever the current method gets from `self.cleaned_data`? It feels like the form shouldn't be the one doing this? Feel free to dispute or ignore this comment.

Reply:
Yes, this is weird, you are right. I think it should actually have been calling this from the form.save method. I've updated it.

Reply:
https://github.com/dimagi/open-chat-studio/pull/2324/commits/66920dbf294b7515a6040c05fc659f80950952f6